libtvdb/__init__.py

```diff
@@ -208,20 +208,28 @@ def get_paged(self, url_path: str, *, timeout: float) -> List[Any]:
 
         return all_results
 
-    def search_show(self, show_name: str, *, timeout: float = 10.0) -> List[Show]:
+    def search_show(self, show_name: str = None, imdb_name: str = None, *, timeout: float = 10.0) -> List[Show]:
         """Search for shows matching the name supplied.
 
         If no matching show is found, a NotFoundException will be thrown.
         """
 
-        if show_name is None or show_name == "":
+        if not show_name and not imdb_name:
             return []
 
-        encoded_name = urllib.parse.quote(show_name)
+        params = encoded_name = ""
 
-        Log.info(f"Searching for show: {show_name}")
+        if show_name:
+            params = "name={}".format(show_name)
```

Comment:
I'd make `params` a dictionary that's empty, then add the correct parameter and use `urllib.parse.urlencode(params)` and append that to the URL.